### DISCUSSION TITLE
v5 updateOrCreate, updateOrCreateMany multiple criteria properties

### DISCUSSION THREAD

**Discussion by skrenek** [adonisjs/core#2894] [Help] [answered]
After reading the docs and double-checking intellisense, it appears Lucid's updateOrCreate (and its "Many" counterpart) method only accepts a single search criteria property for the lookup part.  Is there a reason why it could not accept multiple criteria properties?  

I have a use case where I would like to update or create records whose uniqueness applies to a foreign key and integer property on the model.  For example, to borrow from [the documentation's ](https://docs.adonisjs.com/reference/orm/base-model#static-updateorcreatemany) User sample, let's assume I added a companyId foreign key property to the User model and wanted to updateOrCreateMany based on companyId and email.

**Comment by shunnmugam** [comment, marked as answer]
as per [model](https://github.com/adonisjs/lucid/blob/f1c2e5fad26337086dc188061f97d64463e00764/src/Orm/BaseModel/index.ts#L890), you can pass array of string

`User.updateOrCreateMany(['email','companyId'],users)`

**Reply by skrenek** [reply]
I completely missed that in the typescript signature, and the docs only give an example of a single.  Thank you for pointing me in the right direction. 😃
